sympy/polys/rings.py

```diff
@@ -2531,20 +2531,327 @@ def compose(f, x, a=None):
+        that are polynomials, such as `R[x,y][z]`. The pseudo-quotient `q` and
+        pseudo-remainder `r` with respect to `z` when dividing `f` by `g`
+        satisfy `mf = gq + r`, where `deg(r,z) < deg(g,z)` and
+        `m = LC(g,z)**(deg(f,z) - deg(g,z)+1)`.
+
+        The `prem` method returns the pseudo-remainder `r`,
+        the `pquo` method returns the pseudo-quotient `q`,
+        and `pdiv` returns the tuple `(q, r)`.
+
+        Parameters
+        ==========
+
+        g : PolyElement
+            The polynomial to divide `self` by.
+        x : generator or generator index, optional
+            The main variable of the polynomials and default is first generator.
+
+        Returns
+        =======
+
+        PolyElement
+            The pseudo-remainder polynomial.
 
-    def pdiv(f, g):
-        return f.ring.dmp_pdiv(f, g)
+        Raises
+        ======
 
-    def prem(f, g):
-        return f.ring.dmp_prem(f, g)
+        ZeroDivisionError : If `g` is the zero polynomial.
 
-    def pquo(f, g):
-        return f.ring.dmp_quo(f, g)
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> R, x, y = ring("x, y", ZZ)
+        >>> f = x**2 + x*y
+        >>> g = 2*x + 2
+        >>> f.prem(g) # first generator is chosen by default if it is not given
+        -4*y + 4
+        >>> f.prem(g, y) # generator is given
+        0
+        >>> f.prem(g, 1) # generator index is given
+        0
+
+        See Also
+        ========
+
+        pdiv, pquo, pexquo
+
+        """
+        f = self
+        x = f.ring.index(x)
+        df = f.degree(x)
+        dg = g.degree(x)
+
+        if dg < 0:
+            raise ZeroDivisionError('polynomial division')
+
+        r, dr = f, df
+
+        if df < dg:
+            return r
+
+        N = df - dg + 1
+
+        lc_g = g.coeff_wrt(x, dg)
+
+        xp = f.ring.gens[x]
+
+        while True:
+
+            lc_r = r.coeff_wrt(x, dr)
+            j, N = dr - dg, N - 1
+
+            R = r * lc_g
+            G = g * lc_r * xp**j
+            r = R - G
+
+            dr = r.degree(x)
+
+            if dr < dg:
+                break
+
+        c = lc_g ** N
+
+        return r * c
+
+    def pdiv(self, g, x=None):
+        """
+        Computes the pseudo-division of the polynomial ``self`` with respect to ``g``.
+
+        The pseudo-division algorithm is used to find the pseudo-quotient `q`
+        and pseudo-remainder `r` such that `mf = gq + r`, where `m` represents
+        the multiplier and `f` is the dividend polynomial. The pseudo-quotient `q`
+        and pseudo-remainder `r` are polynomials in the variable `x`, with the
+        degree of `r` with respect to `x` being strictly less than the degree of `g`
+        with respect to `x`.
+        The multiplier `m` is defined as `LC(g, x) ^ (deg(f, x) - deg(g, x) + 1)`,
+        where `LC(g, x)` represents the leading coefficient of `g`.
```

Comment:
Single backticks in sphinx lead to LaTeX rendering. I expect that
```
`LC(g, x) ^ (deg(f, x) - deg(g, x) + 1)`
```
will render badly. Instead use double backticks for code rendering and dollars for LaTeX but make sure that what you write is something that should either render with LaTeX or be read as code e.g.:
```
$\mathrm{LC}(g, x)^{\mathrm{deg}(f, x) - \mathrm{deg}(g, x) + 1}$
   or
``LC(g, x) ** (degree(f, x) - degree(g, x) + 1)``
```
In some cases it is better to use code rendering and in others it is better to use LaTeX rendering but if you let sphinx apply LaTeX rendering for something that is supposed to be code it will come out weird.

Probably here `LC(g, x)^(defg...` will result in having a superscript bracket which is not what you want.

Reply:
Also the paragraph formatting above is off. Paragraphs should be separated by a blank line. If this last sentence is part of the same paragraph then it should be linear wrapped (remove the newline after full stop).

Reply:
The sphinx job failed. You should try building the docs yourself and also check what they look like and how they render:
https://docs.sympy.org/latest/contributing/build-docs.html